core/src/main/java/org/apache/accumulo/core/fate/accumulo/FateMutatorImpl.java

```diff
@@ -132,12 +142,29 @@ public FateMutator<T> delete() {
     return this;
   }
 
+  /**
```

Comment:
The status column is expected to exist for the lifetime of a fate operation.  As a start of using conditional mutations we could build something around and always checking the status column when an update is made.  This will check that row is in a minimally acceptable state for making an update.  We could add more conditions later.

Could add a method like the following to FateMutatorImpl

```java
/**
 * Requires the status column to be set to one of the passed in statuses in order for an update to be made.  If no statuses are passed in, then the status column must be absent in order for an update to be made.
 */
 public FateMutator<T> requireStatus(TStatus ... statuses) {}
```

With a method like the above could do the following when adding repo.

```java
      FateMutator<T> fateMutator = newMutator(tid).requireStatus(SUBMITTED, IN_PROGRESS);
      fateMutator.putRepo(top.map(t -> t + 1).orElse(1), repo).mutate();
```

When adding a completely new fate tx could do the following which would require the status column to be absent.

```java
  @Override
  public long create() {
    long tid = RANDOM.get().nextLong() & 0x7fffffffffffffffL;

    newMutator(tid).requireStatus().putStatus(TStatus.NEW).putCreateTime(System.currentTimeMillis()).mutate();

    return tid;
  }
```

Reply:
If I am understanding things correctly, this new method would check the value of the status column. Is it possible to have a set of acceptable values for a Condition? To me it seems like you can only set one value with `Condition.setValue()`.

Reply:
To to check for a set of possible values can do the following

 * Set an iterator on the condition that maps the value to something like  `present` or `not present` depending on if the value is in the set.
 * Set the expected value on the condition to `present`

Reply:
Okay I added this in e5fa0da. I think there is an issue though but I can't figure out whats causing it. See my next comment please.